### PR TITLE
refactoring rendering components

### PR DESCRIPTION
**Pull request by jeetiss**

### CODE REVIEW

**Comment by RobinMalfait** [comment]
Nice, I just wanted to make the same PR!

File: src/utils/renderComponentWithRenderProps.tsx [hunk 1]
```diff
@@ -0,0 +1,30 @@
+import * as React from 'react';
+import { SharedRenderProps } from '../types';
+import { isEmptyChildren } from './isEmptyChildren';
+
+export function renderComponentWithRenderProps<S>(
+  props: SharedRenderProps<S>,
+  state: S
+) {
+  const { render, component, children } = props;
```
Comment (Pyrolistical) [inline src/utils/renderComponentWithRenderProps.tsx]: inline to line 6?

**Comment by JReinhold** [comment]
Looks good, I think this should be merged.

**Comment by jaredpalmer** [comment]
Had to revert, this needs a generic to not break TS types

**Pull request merged**